packages/truffle-contract/lib/contract/constructorMethods.js

```diff
@@ -1,391 +1,305 @@
+    if (this.network_id && this.networks[this.network_id] != null) {
+      try {
+        const { gasLimit } = await this.web3.eth.getBlock("latest");
+        return { id: this.network_id, blockLimit: gasLimit };
+      } catch (error) {
+        throw error;
       }
-
-      var args = Array.prototype.slice.call(arguments);
-
-      // Promievent and flag that allows instance to resolve (rather than just receipt)
-      var context = {
-        contract: constructor,
-        promiEvent: promiEvent,
-        onlyEmitReceipt: true
-      };
-
-      constructor
-        .detectNetwork()
-        .then(network => {
-          utils.checkLibraries.apply(constructor);
-          return execute.deploy.call(
-            constructor,
-            args,
-            context,
-            network.blockLimit
-          );
-        })
-        .catch(promiEvent.reject);
-
-      return promiEvent.eventEmitter;
-    },
-
-    at: function(address) {
-      var constructor = this;
-
-      return new Promise(function(accept, reject) {
-        if (
-          address == null ||
-          typeof address !== "string" ||
-          address.length !== 42
-        ) {
-          var err =
-            "Invalid address passed to " +
-            constructor.contractName +
-            ".at(): " +
-            address;
-          reject(new Error(err));
-        }
-
-        return constructor.detectNetwork().then(function() {
-          var instance = new constructor(address);
-
-          return constructor.web3.eth.getCode(address).then(function(code) {
-            if (!code || code.replace("0x", "").replace(/0/g, "") === "") {
-              var err =
-                "Cannot create instance of " +
-                constructor.contractName +
-                "; no code at address " +
-                address;
-              reject(new Error(err));
-            }
-
-            accept(instance);
-          });
-        });
-      });
-    },
-
-    deployed: function() {
-      var constructor = this;
-      return constructor.detectNetwork().then(function() {
-        // We don't have a network config for the one we found
-        if (constructor._json.networks[constructor.network_id] == null) {
-          var error =
-            constructor.contractName +
-            " has not been deployed to detected network" +
-            " (network/artifact mismatch)";
-          throw new Error(error);
-        }
-
-        // If we found the network but it's not deployed
-        if (!constructor.isDeployed()) {
-          var error =
-            constructor.contractName +
-            " has not been deployed to detected network (" +
-            constructor.network_id +
-            ")";
-
-          throw new Error(error);
-        }
-
-        return new constructor(constructor.address);
-      });
-    },
-
-    defaults: function(class_defaults) {
-      if (this.class_defaults == null) {
-        this.class_defaults = {};
+    } else {
```

Comment:
You can get rid of the `else` since every branch above either `return`s or `throw`s. Just if you want to remove an indent.